x/trigger/client/cli/tx.go

```diff
@@ -0,0 +1,261 @@
+			if err != nil {
+				return fmt.Errorf("unable to parse time (%v) required format is RFC3339 (%v) , %w", args[0], time.RFC3339, err)
+			}
+
+			msgs, err := parseTransactions(clientCtx.Codec, args[1])
+			if err != nil {
+				return fmt.Errorf("unable to parse message file")
+			}
+			if len(msgs) == 0 {
+				return fmt.Errorf("no actions added to trigger")
+			}
+
+			msg := types.NewCreateTriggerRequest(
+				callerAddr.String(),
+				&types.BlockTimeEvent{Time: startTime.UTC()},
+				msgs,
+			)
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+// GetCmdDestroyTrigger is a command to destroy an existing trigger.
+func GetCmdDestroyTrigger() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "destroy-trigger [id]",
+		Args:    cobra.ExactArgs(1),
+		Aliases: []string{"destroy", "d"},
+		Short:   "Destroys an existing trigger.",
+		Long:    strings.TrimSpace(`Destroys an existing trigger. The trigger will not be destroyable if it has already been detected.`),
+		Example: fmt.Sprintf(`$ %[1]s tx trigger destroy-trigger 1`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			callerAddr := clientCtx.GetFromAddress()
+			triggerID, err := strconv.Atoi(args[0])
+			if err != nil {
+				return fmt.Errorf("invalid trigger id: %s", args[0])
+			}
+
+			msg := types.NewDestroyTriggerRequest(
+				callerAddr.String(),
+				uint64(triggerID),
+			)
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+// parseTransactions reads and parses the message.
+func parseTransactions(cdc codec.Codec, path string) ([]sdk.Msg, error) {
+	contents, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var action ParsedAction
+	err = json.Unmarshal(contents, &action)
+	if err != nil {
+		return nil, err
+	}
+
+	var msg sdk.Msg
+	err = cdc.UnmarshalInterfaceJSON(action.Action, &msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return []sdk.Msg{msg}, nil
+}
+
+// parseEvent reads and parses the transaction event from a file.
+func parseEvent(path string) (*types.TransactionEvent, error) {
+	contents, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var event ParsedTransactionEvent
+	err = json.Unmarshal(contents, &event)
+	if err != nil {
+		return nil, err
+	}
+
+	newEvent := types.TransactionEvent{
+		Name: event.Name,
+	}
+	for _, attr := range event.Attributes {
+		newEvent.Attributes = append(newEvent.Attributes, types.Attribute{
+			Name:  attr.Name,
+			Value: attr.Value,
+		})
+	}
+
+	return &newEvent, nil
+}
+
+// ParsedAction is the deserialized form of the inputted JSON formatted sdk.Msg.
+type ParsedAction struct {
+	// Action defines a sdk.Msg proto-JSON-encoded as Any.
+	Action json.RawMessage `json:"message,omitempty"`
+}
```

Comment:
It'd be easier for folks to use if they can just provide the message directly instead of having to wrap in an object's "action" field.

The `tx` commands can be used to easily create message JSON. But to then provide that to this command, you'd have to put that json in `{"action":...}`.

I'm unfamiliar with `UnmarshalInterfaceJSON` but you should (hopefully) be able to just provide `contents` to it directly to convert it to the `Msg`. If, for some reason, it needs the pre-processing step of being first converted to a `json.RawMessage`, you'd just do `var rawMsg json.RawMessage; err = json.Unmarshal(contents, &rawMsg)`, then provide `rawMsg` to `UnmarshalInterfaceJSON`.

Reply:
Agreed, that is annoying.